File: codeWars/moving_zeros_to_the_end/moving_zeros_to_the_end.py
# Moving Zeros To The End

def moveZeros(array):
    zerosCount = 0
    length = len(array)

    for i in range(length):
        if array[i] != 0:
            array[i - zerosCount] = array[i]
        else:
            zerosCount += 1

    for i in range(length - zerosCount, length):
        array[i] = 0

    return array


moveZ = moveZeros([False, 1, 0, 1, 2, 0, 1, 3, "a"])
print(moveZ)

File: codeWars/katas/moving_zeros_to_the_end/moving_zeros_to_the_end.js
// Moving Zeros To The End

function moveZeros(array) {
    let zerosCount = 0;
    const length = array.length;

    for (let i = 0; i < length; i++) {
        if (array[i] !== 0) {
            array[i - zerosCount] = array[i];
        } else {
            zerosCount++;
        }
    }

    for (let i = length - zerosCount; i < length; i++) {
        array[i] = 0;
    }

    return array;
}

const moveZ = moveZeros([false, 1, 0, 1, 2, 0, 1, 3, "a"]);
console.log(moveZ)